libcontainer/cgroups/fs/freezer.go

```diff
@@ -28,33 +28,49 @@ func (s *FreezerGroup) Apply(path string, d *cgroupData) error {
 
 func (s *FreezerGroup) Set(path string, cgroup *configs.Cgroup) error {
 	switch cgroup.Resources.Freezer {
-	case configs.Frozen, configs.Thawed:
-		for {
-			// In case this loop does not exit because it doesn't get the expected
-			// state, let's write again this state, hoping it's going to be properly
-			// set this time. Otherwise, this loop could run infinitely, waiting for
-			// a state change that would never happen.
-			if err := fscommon.WriteFile(path, "freezer.state", string(cgroup.Resources.Freezer)); err != nil {
+	case configs.Frozen:
+		// As per kernel doc (freezer-subsystem.txt), if FREEZING
+		// is seen, userspace should either retry or thaw.
```

Comment:
It should be noted that the comment describing this behaviour was removed from the kernel documentation in 2012 (torvalds/linux@ef9fe980c6fcc1821ab955b74b242d2d6585fa75) when the freezer cgroup gained hierarchical support. I'm not sure if this was intentional or not but just something to be aware of.

Reply:
Ah! That at least explains why we don't ever see EBUSY.

Comment updated.